test/release/examples/primers/ranges.chpl

```diff
@@ -7,245 +7,384 @@
 /*
   .. _primers-ranges-general:
 
-  Ranges (general info)
-  ---------------------
+  Range Basics
+  ------------
 */
 //
-// Ranges represent sequences of integral numbers, or "indices".
-// A range literal is defined using the ``..`` operator.
-// The lower bound is always on the left;
-// the upper bound is always on the right.
-//
-writeln("Basic range r");
-const r = 1..10;
+// Ranges represent regular sequences of values, such as integers, and
+// are typically defined in terms of a low and high bound.  Range
+// values can be specified using the ``..`` operator, where the low
+// bound is specified on the left side of the ``..`` and the high
+// bound is on the right.  These bounds are inclusive for ranges
+// created with ``..``.
+//
+const n = 5,
+      lo = -3,
+      hi = 3;
+writeln("Basic ranges");
+const r = 1..10,    // 1, 2, 3, ..., 10
+      r2 = 0..n,    // 0, 1, 2, ..., n
+      r3 = lo..hi;  // -3, -2, -1, ..., 3
 writeRange(r);
+writeRange(r2);
+writeRange(r3);
+writeln();
 
 // Ranges can also be defined using an `open-interval` form using the
-// ``..<`` operator which causes the upper bound to be omitted.  For
-// example, the range ``3..<10`` is equivalent to ``3..9``.
-writeln("Open interval range rOpen");
-const rOpen = 3..<10;
+// ``..<`` operator, causing the upper bound to be omitted.  For
+// example, the range ``3..<10`` is equivalent to ``3..9``.  This form
+// is particularly useful for 0-based indexing, to avoid bounds like
+// ``n-1``.
+writeln("Open interval ranges");
+const rOpen = 1..<10,  // 1, 2, 3, ..., 9
+      rOpen2 = 0..<n;  // 0, 1, 2, ..., n-1
 writeRange(rOpen);
+writeRange(rOpen2);
+writeln();
+
+
+// Note that whenever a range's low bound exceeds its high bound, the
+// range is considered to be empty.
+writeln("Empty ranges");
+const empty1 = 1..0,
+      empty2 = 0..-1,
+      empty3 = 10..1;
+writeRange(empty1);
+writeRange(empty2);
+writeRange(empty3);
+writeln();
+
+// To create a range representing a decreasing series of values, the
+// ``by`` operator can be used with a negative stride.  As a simple
+// example, the following range represents the integers counting from
+// 10 down to 1.
```

Comment:
I would probably first talk about negative stride, and then mention that low>high means empty range.

Reply:
I kind of preferred this approach because I think it addresses the thing that people try and fail with first, hopefully driving that message home clearly before losing attention / leading them to the solution.  I also think that ending this little section on a forward reference feels cleaner than putting the forward reference before the note about empty ranges.  Would you be OK with keeping it as-is?

Reply:
Yeah, I am OK with it as-is.